mesheryctl/pkg/utils/scripts.go

```diff
@@ -6,118 +6,252 @@ import (
-			exit;
-		fi
-		if ! command_exists awk ; then
-			echo -e "\\nMissing required utility: 'awk'. Please install 'awk' and try again."
-			exit;
-		fi
-		if ! command_exists tail ; then
-			echo -e "\\nMissing required utility: 'tail'. Please install 'tail' and try again."
-			exit;
-		fi
+
+	return cfg.executeScript()
+}
+
+func (c *GKEConfig) validate() error {
+	if c.ConfigPath == "" || c.SAName == "" || c.Namespace == "" {
+		return ErrInvalidArgument(fmt.Errorf("configPath, SAName, and namespace are required"))
+	}
+	return nil
+}
+
+func (c *GKEConfig) checkPrerequisites() error {
+	var missingCommands []string
+	requiredCommands := []string{"jq", "base64", "awk", "tail"}
+
+	fmt.Println("\nRequired Commands")
+	fmt.Println("--------------")
+
+	for _, cmd := range requiredCommands {
+		if _, err := exec.LookPath(cmd); err != nil {
+			fmt.Printf("!! %s is not available\n", cmd)
+			missingCommands = append(missingCommands, cmd)
+		} else {
+			fmt.Printf("✓ %s is available\n", cmd)
+		}
 	}
-	create_service_account() {
-		echo -e "\\nCreating a service account in ${NAMESPACE} namespace: ${SERVICE_ACCOUNT_NAME}"
-		kubectl create sa "${SERVICE_ACCOUNT_NAME}" --namespace "${NAMESPACE}"
-		kubectl create clusterrolebinding "${SERVICE_ACCOUNT_NAME}" --clusterrole=cluster-admin --serviceaccount=default:"${SERVICE_ACCOUNT_NAME}" --namespace "${NAMESPACE}"
+
+	if len(missingCommands) > 0 {
+		return ErrMissingCommands(fmt.Errorf("missing required commands: %v", missingCommands))
 	}
+	return nil
+}
 
-	get_secret_name_from_service_account() {
-		echo -e "\\nGetting secret of service account ${SERVICE_ACCOUNT_NAME} on ${NAMESPACE}"
-		SECRET_NAME=$(kubectl get sa "${SERVICE_ACCOUNT_NAME}" --namespace="${NAMESPACE}" -o json | jq -r .secrets[].name)
-		echo "Secret name: ${SECRET_NAME}"
+func (c *GKEConfig) executeScript() error {
+	if err := c.checkConnectivity(); err != nil {
+		return err
 	}
 
-	extract_ca_crt_from_secret() {
-		echo -e -n "\\nExtracting ca.crt from secret..."
-		kubectl get secret --namespace "${NAMESPACE}" "${SECRET_NAME}" -o json | jq \
-		-r '.data["ca.crt"]' | base64 --decode > "${TARGET_FOLDER}/ca.crt"
-		printf "done"
+	if err := c.checkPrerequisites(); err != nil {
+		return err
 	}
 
-	get_user_token_from_secret() {
-		echo -e -n "\\nGetting user token from secret..."
-		USER_TOKEN=$(kubectl get secret --namespace "${NAMESPACE}" "${SECRET_NAME}" -o json | jq -r '.data["token"]' | base64 --decode)
-		printf "done"
+	script := c.generateScript()
+	cmd := exec.Command("sh", "-c", script)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func (c *GKEConfig) checkConnectivity() error {
+	fmt.Println("\nKubernetes API")
+	fmt.Println("--------------")
+
+	// Check if client can be initialized
+	cmd := exec.Command("kubectl", "cluster-info")
+	cmd.Stdout = nil
+	cmd.Stderr = nil
+
+	if err := cmd.Run(); err != nil {
+		fmt.Println("!! cannot connect to Kubernetes API server")
```

Comment:
```suggestion
```
There error message should be enough by itself

Reply:
signed off